src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -98,13 +103,32 @@ public void inferenceSentences(
         @NonNull final String modelId,
         @NonNull final List<String> inputText,
         @NonNull final ActionListener<List<List<Float>>> listener
+    ) {
+        inferenceSentencesWithRetry(targetResponseFilters, modelId, inputText, 0, listener);
+    }
+
+    private void inferenceSentencesWithRetry(
+        @NonNull final List<String> targetResponseFilters,
+        @NonNull final String modelId,
+        @NonNull final List<String> inputText,
+        final int retryTime,
+        @NonNull final ActionListener<List<List<Float>>> listener
     ) {
         MLInput mlInput = createMLInput(targetResponseFilters, inputText);
         mlClient.predict(modelId, mlInput, ActionListener.wrap(mlOutput -> {
             final List<List<Float>> vector = buildVectorFromResponse(mlOutput);
             log.debug("Inference Response for input sentence {} is : {} ", inputText, vector);
             listener.onResponse(vector);
-        }, listener::onFailure));
+        }, e -> {
+            final int nodeNotConnectedExceptionIndex = ExceptionUtils.indexOfThrowable(e, NodeNotConnectedException.class);
+            final int nodeDisconnectExceptionIndex = ExceptionUtils.indexOfThrowable(e, NodeDisconnectedException.class);
+            if ((nodeDisconnectExceptionIndex != -1 || nodeNotConnectedExceptionIndex != -1) && retryTime < MAX_RETRY) {
```

Comment:
We should encapsulate this in a different method which can give a boolean response if we should do a retry or not. This will allow us in future to add more retryable exceptions and will also allow us to evolve the conditions for doing the retry.

Reply:
Refactored to a new method.